swan-cern/files/swan_spark_config.py

```diff
@@ -347,20 +347,32 @@ def spark_modify_pod_hook(spawner, pod):
     spark_pod_hook_handler = SwanSparkPodHookHandler(spawner, pod)
     return spark_pod_hook_handler.get_swan_user_pod()
 
-
-def spark_cleanup_services(spawner):
-    """
-    :param spawner: Swan Kubernetes Spawner
-    :type spawner: swanspawner.SwanKubeSpawner
-    
-    """
-    spark_cluster = spawner.user_options[spawner.spark_cluster_field]
-    if spark_cluster and spark_cluster != 'none':
-        username = spawner.user.name
-        spark_ports_service = f"spark-ports-{username}"
-        spawner.log.info('Deleting service %s', spark_ports_service)
-        swan_container_namespace = os.environ.get('POD_NAMESPACE', 'default')
-        spawner.api.delete_namespaced_service(spark_ports_service, swan_container_namespace)
+class SwanSparkPostStopHookHandler(SwanPostStopHookHandler):
+    def post_stop_hook(self, spawner):
+        """
+        :param spawner: Swan Kubernetes Spawner
+        :type spawner: swanspawner.SwanKubeSpawner
+        """
+        super().post_stop_hook(spawner)
+        spark_cluster = spawner.user_options[spawner.spark_cluster_field]
+        if spark_cluster and spark_cluster != 'none':
+            username = spawner.user.name
+            swan_container_namespace = os.environ.get('POD_NAMESPACE', 'default')
+
+            # Delete NodePort service opening ports for the user spark processes
+            spark_ports_service = f"spark-ports-{username}"
+            spawner.log.info('Deleting service %s', spark_ports_service)
+            spawner.api.delete_namespaced_service(spark_ports_service, swan_container_namespace)
+            
+            # Delete Kubernetes Secret with hadoop delegation tokens
+            hadoop_secret_name = f"hadoop-tokens-{username}"
+            swan_container_namespace = os.environ.get('POD_NAMESPACE', 'default')
+            spawner.log.info('Deleting secret %s', hadoop_secret_name)
+            spawner.api.delete_namespaced_secret(hadoop_secret_name, swan_container_namespace)
+
+def spark_post_stop_hook(spawner):
+   spark_post_stop_hook_handler = SwanSparkPostStopHookHandler()
+   spark_post_stop_hook_handler.post_stop_hook(spawner)
 
 c.SwanKubeSpawner.modify_pod_hook = spark_modify_pod_hook
-c.SwanKubeSpawner.post_stop_hook = spark_cleanup_services
\ No newline at end of file
+c.SwanKubeSpawner.post_stop_hook = spark_post_stop_hook
```

Comment:
This is supposed to overwrite the setting done in `swan_config_cern.py` for `SwanKubeSpawner.post_stop_hook`? Which one has precedence?

Reply:
This precedence is I think based on the alphabetic ordering of the files in this config folder [here](https://github.com/swan-cern/swan-charts/blob/1c3895446147849764c5393d984ad3ea6416934b/swan-cern/values.yaml#L87)